pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -427,7 +427,7 @@ func (r *ReconcilePerconaServerMongoDB) reconcileStatefulSet(arbiter bool, cr *a
 		if err != nil {
 			return nil, fmt.Errorf("PodDisruptionBudget for %s: %v", sfs.Name, err)
 		}
-
+		sfs.Spec = sfsSpec
 		sfs.Spec.Replicas = &size
 		sfs.Spec.Template.Spec.Containers = sfsSpec.Template.Spec.Containers
 		sfs.Spec.Template.Spec.Volumes = sfsSpec.Template.Spec.Volumes
```

Comment:
so there is no need in L431-433 anymore. as well as the duplication of the same code (`sfs.Spec = sfsSpec`) in the `if` above (L451). Just put it before `if-else`, so `if-else` is just decides to make create or update call.

Reply:
ok, `sfs.Spec.Replicas = &size` we need. but again, I suggest to move if outside of the `if-else`